### PR TITLE
Adds reverse-string exercise

### PR DESCRIPTION
**Pull request by amscotti**

### CODE REVIEW

File: exercises/reverse-string/example.nim [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import
+  algorithm, sequtils, strutils
+
+proc reverse*(value: string): string =
+  algorithm.reversed(value).mapIt(string, $it).join
```
Comment (dom96) [inline exercises/reverse-string/example.nim]: Isn't this cheating a bit? The mapIt also looks redundant to me.

Comment (amscotti) [reply exercises/reverse-string/example.nim]: @dom96 I don't feel this is cheating, as the read me doesn't say how to solve this exercise but I will work on a more fitting solution when I have a bit more time.

As far as the `mapIt`, I would love a better way of  doing this without the `mapIt`. The output from `algorithm.reversed` gave me an sequence of chars, which I couldn't find a better way to to get a string out of it them without first mapping them to strings and then joining them. Overall this did seem odd to me but I couldn't find a better way. Got any tips on this?

Comment (SolitudeSF) [reply exercises/reverse-string/example.nim]: `join` overload for nonstrings calls `$` itself.

Comment (amscotti) [reply exercises/reverse-string/example.nim]: @SolitudeSF You are right, I missed that. Thanks for pointing this out to me!

**Comment by amscotti** [comment]
The code has been updated to do the reverse itself and not using Nim's module for algorithms, but this exercise could also be solved in this way,

```nim
import
  algorithm, strutils

proc reverse*(value: string): string =
  value.reversed.join
```

**Pull request merged**